mm2src/mm2_rpc/src/data/legacy.rs

```diff
@@ -254,4 +256,95 @@ pub struct MmVersionResponse {
     pub datetime: String,
 }
 
-fn get_true() -> bool { true }
+#[derive(Debug, Deserialize, Serialize)]
+#[serde(tag = "method", rename_all = "lowercase")]
+pub enum ActivationRequest {
+    Enable(EnableRequest),
+    Electrum(ElectrumRequest),
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+pub struct EnableRequest {
+    pub coin: String,
+    #[serde(default, skip_serializing_if = "Vec::is_empty")]
+    pub urls: Vec<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub swap_contract_address: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub fallback_swap_contract: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub gas_station_url: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub gas_station_decimals: Option<u8>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub gas_station_policy: Option<GasStationPricePolicy>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub mm2: Option<u8>,
+    #[serde(default)]
+    pub tx_history: bool,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub required_confirmations: Option<u64>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub requires_notarization: Option<bool>,
+}
+
+/// Using tagged representation to allow adding variants with coefficients, percentage, etc in the future.
+#[derive(Clone, Debug, Deserialize, Serialize)]
+#[serde(tag = "policy", content = "additional_data")]
+pub enum GasStationPricePolicy {
+    /// Use mean between average and fast values, default and recommended to use on ETH mainnet due to
+    /// gas price big spikes.
+    MeanAverageFast,
+    /// Use average value only. Useful for non-heavily congested networks (Matic, etc.)
+    Average,
+}
+
+impl Default for GasStationPricePolicy {
+    fn default() -> Self { GasStationPricePolicy::MeanAverageFast }
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+pub struct ElectrumRequest {
+    pub coin: String,
+    #[serde(default, skip_serializing_if = "Vec::is_empty")]
+    pub servers: Vec<Server>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub mm2: Option<u8>,
+    #[serde(default)]
+    pub tx_history: bool,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub required_confirmations: Option<u64>,
+    #[serde(default)]
+    pub requires_notarization: bool,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub swap_contract_address: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub fallback_swap_contract: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub utxo_merge_params: Option<UtxoMergeParams>,
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+pub struct Server {
+    pub url: String,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub protocol: Option<Protocol>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub disable_cert_verification: Option<bool>,
```

Comment:
`Default` should probably be used here if `protocol` or `disable_cert_verification` were not passed. https://github.com/KomodoPlatform/komodo-defi-framework/blob/1c261cd5a39694ae13f2699a5c4761df3468e19d/mm2src/coins/utxo/rpc_clients.rs#L1419-L1427

Reply:
Sure, thank you for pointing it out.
[done](https://github.com/KomodoPlatform/komodo-defi-framework/pull/1912/commits/92a429ce87c27b1780015cfa2d7741950f912f7a#diff-a0c7b4a6ee8f58e5565de61f8969cfa419a81e76572ca50ef8cab8121751e89cR64-R71)